src/Shared/LayoutRenderers/AspNetResponseCookieLayoutRenderer.cs

```diff
@@ -0,0 +1,194 @@
+            if (httpResponse == null)
+            {
+                return;
+            }
+
+            var cookies = GetCookies(httpResponse);
+
+            if (cookies?.Count > 0)
+            {
+                bool checkForExclude = (CookieNames == null || CookieNames.Count == 0) && Exclude?.Count > 0;
+                var cookieValues = GetCookieValues(cookies, checkForExclude);
+                SerializePairs(cookieValues, builder, logEvent);
+            }
+        }
+
+#if !ASP_NET_CORE
+
+        /// <summary>
+        /// Method to wrap getting cookies from the HTTP Response for both Framework and Core
+        /// </summary>
+        /// <param name="response"></param>
+        /// <returns></returns>
+        protected Cookies GetCookies(HttpResponseBase response)
+        {
+            return response.Cookies;
+        }
+
+        private List<string> GetCookieNames(HttpCookieCollection cookies)
+        {
+            return CookieNames?.Count > 0 ? CookieNames : cookies.Keys.Cast<string>().ToList();
+        }
+
+        private IEnumerable<KeyValuePair<string, string>> GetCookieValues(HttpCookieCollection cookies, bool checkForExclude)
+        {
+            var cookieNames = GetCookieNames(cookies);
+            foreach (var cookieName in cookieNames)
+            {
+                if (checkForExclude && Exclude.Contains(cookieName))
+                    continue;
+
+                var httpCookie = cookies[cookieName];
+                if (httpCookie == null)
+                {
+                    continue;
+                }
+
+                if (OutputFormat != AspNetRequestLayoutOutputFormat.Flat)
+                {
+                    // Split multi-valued cookie, as allowed for in the HttpCookie API for backwards compatibility with classic ASP
+                    var isFirst = true;
+                    foreach (var multiValueKey in httpCookie.Values.AllKeys)
+                    {
+                        var cookieKey = multiValueKey;
+                        if (isFirst)
+                        {
+                            cookieKey = cookieName;
+                            isFirst = false;
+                        }
+                        yield return new KeyValuePair<string, string>(cookieKey, httpCookie.Values[multiValueKey]);
+                    }
+                }
+                else
+                {
+                    yield return new KeyValuePair<string, string>(cookieName, httpCookie.Value);
+                }
+            }
+        }
+
+#elif ASP_NET_CORE2
+
+        /// <summary>
+        /// Method to wrap getting cookies from the HTTP Response for both Framework and Core
+        /// </summary>
+        /// <param name="response"></param>
+        /// <returns></returns>
+        protected IList<SetCookieHeaderValue> GetCookies(HttpResponse response)
+        {
+            var queryResults = response.Headers.Where(row => row.Key == "Set-Cookie").ToList();
```

Comment:
Can one use `Microsoft.Net.Http.Headers.HeaderNames.SetCookie` ?

Reply:
If `response.Headers` is type `IHeaderDictionary`. Can one not use `IHeaderDictionary.SetCookie`-property ?

Reply:
So one just need to convert `SetCookie`-`StringValues` to something that `SetCookieHeaderValue.TryParse` can handle.

Reply:
yes that is what I am working on now

Reply:
I found IHeaderDictionary.SetCookie not exist in net std 2.0/net 461 case.

Reply:
so i am working around that